src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/db/RecallRepository.kt

```diff
@@ -29,6 +29,6 @@ class RecallRepository(
     try {
       getById(recallId.value)
     } catch (e: JpaObjectRetrievalFailureException) {
-      throw RecallNotFoundException("Recall not found: '$recallId'", e)
```

Comment:
Don't see any issue about being explicit with an error message.

Reply:
Seems a lot simpler to just contain the relevant ID of the object not found.  When we log it the type of the exception is logged with the recallId which explains clearly what has not been found.  e.g.
`2021-09-02 15:39:33.979  INFO 47841 --- [o-auto-1-exec-8] d.h.m.c.ManageRecallsApiExceptionHandler : RecallNotFoundException(recallId=163abd37-11d7-4954-8de2-ee8ed9c7c98f)`

Reply:
To me a type that clearly explains what its purpose is and to what it relates is much simpler, and much easier to search through the logs in a consistent manner, a random message string is more prone to be different in different NotFoundException subclasses and hence harder to do more complicated log queries that span more than one exception type (if ever needed).

Reply:
I guess it depends how you look at the logs ; when I use Kibana for example I will generally just look at the log level and the error message. This format forces you to also display the stack trace to see the type of the exception, doesn't it?

Reply:
The error message in this case would be: 
**RecallNotFoundException(recallId=163abd37-11d7-4954-8de2-ee8ed9c7c98f)** 
Just the toString of the RecallNotFoundException data class.  It would be better if we were logging them as json, more structured logging makes complicated querying/analysis of logs a lot easier